edb_stat_statements/edb_stat_statements.c

```diff
@@ -608,14 +603,31 @@ edbss_extract_stmt_info(const char* query_str, int query_len) {
 				.object_field_start = edbss_json_ofield_start,
 				.scalar = edbss_json_scalar,
 		};
-		JsonParseErrorType parse_rv = pg_parse_json(lex, &sem);
-		freeJsonLexContext(lex);
-
-		if (parse_rv == JSON_SUCCESS)
-			if ((state.found & EDB_STMT_INFO_PARSE_REQUIRED) == EDB_STMT_INFO_PARSE_REQUIRED)
-				if (info->query_id != UINT64CONST(0))
-					return info;
 
+		while (info_str) {
+			JsonLexContext *lex = makeJsonLexContextCstringLen(
+#if PG_VERSION_NUM >= 170000
+					NULL,
+					info_str,
+#else
+					(char *) info_str,  // not actually mutating
+#endif
+					info_len,
+					PG_UTF8,
+					true);
+			JsonParseErrorType parse_rv = pg_parse_json(lex, &sem);
+			freeJsonLexContext(lex);
+
+			if (parse_rv == JSON_SUCCESS)
+				if ((state.found & EDB_STMT_INFO_PARSE_REQUIRED) == EDB_STMT_INFO_PARSE_REQUIRED)
+					return info->query_id != UINT64CONST(0) ? info : NULL;
+
+			info_str += info_len + 1;
```

Comment:
The +1 makes me nervous about the case where there isn't a newline at the end of an info line?

Reply:
I'm not sure if the cases where there are untrusted entries in the query log is that likely, but this is C so I want to be extra careful about our boundary cases.

Reply:
Right, good question. The `edbss_extract_info_line()` function will tag the `\n` or the end of the `query_str`, so this `+1` will either skip the `\n` properly, or go beyond the end of the `query_str` and cause the next call to `edbss_extract_info_line()` to return `NULL`.

Reply:
Because len is negative, at that point?

Reply:
Yes exactly